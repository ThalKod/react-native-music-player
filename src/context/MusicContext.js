import React from "react";

const MusicContext = React.createContext({
  selected: {},
  list: [],
  changeSelected: () => {}
});

export const MusicProvider = MusicContext.Provider;
export const MusicConsumer = MusicContext.Consumer;

export default MusicContext;
