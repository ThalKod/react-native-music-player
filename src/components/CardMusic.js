import React, { useContext } from "react";
import {
  View,
    Text,
    StyleSheet,
    Image
} from "react-native";

import defaultImg from "../../assets/images/default_music_cover.png";
import MusicContext from "../context/MusicContext";
import { moderateScale } from "react-native-size-matters";

import { color } from "../config";

const CardMusic = () => {
  const { selected = {} } = useContext(MusicContext);

  const cutText = (string) => {
    let newString = string.substring(0,25);
    if(string.length >= 25) newString += "...";
    return  newString;
  };

  return(
      <View style={styles.container}>
        <Image
              style={{
                alignSelf: 'center',
                height: "75%",
                width: "80%",
                borderRadius: moderateScale(15)
              }}
              source={selected.cover ? {uri: selected.cover } : defaultImg}
          />
          {Object.keys(selected).length ? (
              <View style={styles.description}>
                <Text style={styles.title}>{selected.title ? cutText(selected.title) : cutText(selected.fileName)}</Text>
                <Text style={[styles.title, { fontSize: moderateScale(24)}]}>{selected.author ? cutText(selected.author) : "Unknow Artist"}</Text>
              </View>
          ) : <Text>No music selected</Text>}
      </View>
  )
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    width: "100%",
    height: "65%",
    paddingTop: "5%"
  },
  description: {
    marginTop: "10%",
    alignItems: "center",
  },
  title: {
    fontSize: moderateScale(28),
    fontFamily: "Avenir-Book",
    color: color.white
  }
});

export default CardMusic;
