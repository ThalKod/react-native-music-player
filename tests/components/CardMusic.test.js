import React from "react";
import { create } from 'react-test-renderer';
import { musicData } from "../__mock__/index";
import MusicContext from "../../src/context/MusicContext";
import CardMusic from "../../src/components/CardMusic";
import defaultImg from "../../assets/images/default_music_cover.png";


describe("CardMusic component", () => {
  test("Should correctly render the component", () => {
    const context = {
      selected: musicData[0]
    };
    const tree = create(
        <MusicContext.Provider value={context}>
          <CardMusic/>
        </MusicContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });

  test("Should render default image and text if music not selected", () => {
    const context = {
        selected: {}
    };

    const component = create(
        <MusicContext.Provider value={context}>
          <CardMusic/>
        </MusicContext.Provider>
    );

    const root = component.root;
    const image  = root.findByType("Image");
    const text = root.findByType("Text");

    expect(text.props.children).toBe("No music selected");
    expect(image.props.source).toBe(defaultImg);
  })
});
