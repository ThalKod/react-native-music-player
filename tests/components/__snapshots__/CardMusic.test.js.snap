// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMusic component Should correctly render the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "65%",
      "paddingTop": "5%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/default_music_cover.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 23.57142857142857,
        "height": "75%",
        "width": "80%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": "10%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir-Book",
          "fontSize": 44,
        }
      }
    >
      Test music title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir-Book",
            "fontSize": 44,
          },
          Object {
            "fontSize": 37.714285714285715,
          },
        ]
      }
    >
      Unknow Artist
    </Text>
  </View>
</View>
`;
